### ISSUE TITLE
Ability to see the messages sent

### ISSUE THREAD

**Issue by dcodesdev** [knadh/listmonk#2185]
### Description

Currently, the campaigns only show how many emails have been sent (e.g. 140/1000) but there is no way to know, who received it and who didn't?

It would be nice to have another tab just like we have to subscribers, campaigns, etc. But this one for messages and have all the details specific to that email (sent at, to whom, etc.)

This feature can be disabled by default so that users don't get surprised with their database being full all of a sudden.

**Comment by MaximilianKohler** [comment]
Yes, I think this would be nice and relatively easy, since listmonk sends out the email in reverse order of the list (_IE: go to the last page of the subscriber list for the campaign, and it sends to the last subscriber first_).

**Comment by knadh** [comment]
> Yes, I think this would be nice and relatively easy, since listmonk sends out the email in reverse order of the list 

This wouldn't capture e-mails that failed in the middle though.

**Comment by MaximilianKohler** [comment]
>This wouldn't capture e-mails that failed in the middle though.

Wouldn't it be fairly simple to mark or subtract the failed ones? They're already logged so it seems like it would be simple.

**Comment by AlejandroRodino** [comment]
@dcodesdev did you find a solution for this?

**Comment by dcodesdev** [comment]
@AlejandroRodino I use postal as my SMTP server, I can see the messages there, but on ListMonk I can not see them as there's no workaround. This is a feature that can be implemented by the ListMonk team, and I think it is necessary for it to have a separate database table for the messages that save all the data related to that message, including the recipient, the final rendered message HTML, timestamps, bounced, etc.

**Comment by MaximilianKohler** [comment]
>I think it is necessary for it to have a separate database table for the messages that save all the data related to that message, including the recipient, the final rendered message HTML, timestamps, bounced, etc.

I think it hasn't been done because it would greatly increase the size of the database and they're trying to keep it lightweight. But an option to enable/disable would give people the choice themselves. It could include a warning that enabling it greatly increases the database size.

**Comment by cbertelli** [comment]
At the moment this is very important for me because I send messages with multiple accounts/servers and some of them produce unexpected errors. 
Putting everything on the database may be heavy, but Listmonk provides a Maintenance menu that could host a section about sent messages.
Another option could be using the log. It's not so convenient, but it can be deleted after some time and does not burder the database.

**Comment by phylor** [comment]
Only posting to remove the stale label.

Knowing which recipients an email was sent to (i.e. handed successfully off to an SMTP server) is crucial. This allows resending the failed ones. I can't quite imagine how this is done if you have millions of subscribers. But even with hundreds or thousands it's quite cumbersome to skim the logs.

Storing which recipient was sent a campaign (with state successful/failed) in the database would be great. This would also allow to have a history of sent campaigns per subscriber.

**Comment by cbertelli** [comment]
My not-so-smart hidden expectation is that having a complete tracing of the
sent messages, I could set a dynamic relationship with our database. Even
if my dynamic recipient list changes, I could keep a trace of the messages.

**Comment by dcodesdev** [comment]
@AlejandroRodino @cbertelli @phylor I didn't find a solution with ListMonk, but built my own OS app that solves this problem https://github.com/dcodesdev/LetterSpace
